core/src/test/java/io/questdb/test/cutlass/text/TextLoaderTest.java

```diff
@@ -2677,6 +2680,64 @@ public void testUtf8Tab() throws Exception {
         });
     }
 
+    @Test
+    public void testWriteAfterColumnWasRecreated() throws Exception {
+        assertNoLeak(textLoader -> {
+            final String csv = "col_a,col_b\n" +
+                    "1,2\n" +
+                    "1,2\n";
+            final String expected = "col_b\tcol_a\n" +
+                    "2\t1\n" +
+                    "2\t1\n";
+
+            compiler.compile("create table test (col_a int, col_b long)", sqlExecutionContext);
+            compile(compiler, "alter table test drop column col_a", sqlExecutionContext);
+            compile(compiler, "alter table test add column col_a long", sqlExecutionContext);
+
+            configureLoaderDefaults(textLoader);
+            playText(
+                    textLoader,
+                    csv,
+                    1024,
+                    expected,
+                    "{\"columnCount\":3,\"columns\":[{\"index\":0,\"name\":\"col_a\",\"type\":\"unknown\"},{\"index\":1,\"name\":\"col_b\",\"type\":\"LONG\"},{\"index\":2,\"name\":\"col_a\",\"type\":\"LONG\"}],\"timestampIndex\":-1}",
+                    2,
+                    2
+            );
+        });
+    }
+
+    @Test
+    public void testWriteAfterColumnWasRecreatedNoHeader() throws Exception {
+        assertNoLeak(textLoader -> {
+            final String csv = "1,2\n" +
+                    "1,2\n";
+            final String expected = "col_b\tcol_a\n" +
+                    "2\t1\n" +
+                    "2\t1\n";
+
+            compiler.compile("create table test (col_a int, col_b long)", sqlExecutionContext);
+            compile(compiler, "alter table test drop column col_a", sqlExecutionContext);
+            compile(compiler, "alter table test add column col_a long", sqlExecutionContext);
+
+            configureLoaderDefaults(textLoader);
+            try {
+                playText(
+                        textLoader,
+                        csv,
+                        1024,
+                        expected,
+                        "{\"columnCount\":3,\"columns\":[{\"index\":0,\"name\":\"col_a\",\"type\":\"unknown\"},{\"index\":1,\"name\":\"col_b\",\"type\":\"LONG\"},{\"index\":2,\"name\":\"col_a\",\"type\":\"LONG\"}],\"timestampIndex\":-1}",
+                        2,
+                        2
+                );
+                Assert.fail();
+            } catch (CairoException e) {
+                TestUtils.assertContains(e.getFlyweightMessage(), "cannot match columns by their positions, please add headers or provide schema [index=0]");
```

Comment:
This test should upload without error. After all, the alters the table has 2 columns of type long, int. Uploading CSV of

```
1,2
1,2
```
without header still matches the column

Reply:
The behavior and the tests were updated in 8d9648b31595614da8344512d70fdbe984ec78f3